src/modules/common/containers/LanguageSelector.js

```diff
@@ -1,40 +1,64 @@
 import React from 'react'
 import PropTypes from 'prop-types'
-import { isEmpty } from 'lodash/lang'
 import { connect } from 'react-redux'
+import { Link } from 'redux-little-router'
+import cx from 'classnames'
 import compose from 'lodash/fp/compose'
 
 import withFetcher from 'modules/endpoint/hocs/withFetcher'
 import LANGUAGE_ENDPOINT from 'modules/endpoint/endpoints/languages'
 import LanguageModel from 'modules/endpoint/models/LanguageModel'
 import style from './LanguageSelector.css'
-import cx from 'classnames'
-import { Link } from 'redux-little-router'
 
-class LanguageSelector extends React.Component {
+/**
+ * Displays a dropDown menu to handle changing of the language
+ */
+export class LanguageSelector extends React.Component {
   static propTypes = {
-    languages: PropTypes.arrayOf(PropTypes.instanceOf(LanguageModel)),
-    closeDropDownCallback: PropTypes.func,
-    language: PropTypes.string,
+    closeDropDownCallback: PropTypes.func.isRequired,
+    languages: PropTypes.arrayOf(PropTypes.instanceOf(LanguageModel)).isRequired,
+    language: PropTypes.string.isRequired,
     location: PropTypes.string.isRequired,
     languageChangeUrls: PropTypes.object.isRequired
   }
 
+  /**
+   * Maps the given languageCode to a path to link to, which is either the languageChangeUrl from the store or,
+   * if this is not given, the root categories page in the language
+   * @param languageCode The languageCode
+   * @return {string} The path
+   */
   getPathForLanguage (languageCode) {
     return this.props.languageChangeUrls[languageCode] || `/${this.props.location}/${languageCode}`
   }
 
+  /**
+   * Maps all languages to a Link, which are linking to the same page in a different language and closing the dropDown
+   * menu. For the current language we don't want to link anywhere, so we just close the dropDown menu and prevent the
+   * default behaviour
+   * @return {Link[]} The links for language change
+   */
+  getLanguageLinks () {
+    const preventLink = (event) => {
+      event.preventDefault()
+      this.props.closeDropDownCallback()
+    }
+
+    return this.props.languages.map(language =>
+      <Link
+        key={language.code}
+        className={cx(style.element, this.props.language === language.code ? style.elementActive : '')}
+        onClick={language.code === this.props.language ? preventLink : this.props.closeDropDownCallback}
```

Comment:
Why do we need to prevent the link? We shouldn't do that imo.
We should make closeDropDownCallback not required and check, if closeDropDownCallback is defined, so we can better reuse LanguageSelector

Reply:
what sense does it make to change the language to the same language you're currently at? this does just reload the page, and, because the languageChangeUrls are undefined for the current language even redirects to the root categories page, which definitely shouldn't be the case

Reply:
If you see it like that, then there shouldn't be a <Link /> for the current language in the first place. Maybe you can replace it with a <span />?

Reply:
I think we wanted the links for SEO.
When you click the language your currently having it should just close the popup and do nothing else. 
For formatting move some of the stuff out of the jsx.

Also reformat to this:
className={cx(style.element, {[style.elementActive]: this.props.language === language.code}}

Reply:
Links for other languages are alright, thats good for SEO, but a Link for the current language is not good, right? As steff said, the links for the current language are more or less broken, then a link rather confuses Search engines.

Reply:
Shouldn't we fix the link for the current language then? Or is it not important? If that is the case then we can remove the anchor tag for the current language. The current language acts then as a close button.

Reply:
I think it's not important. And that's what I meant to say :D Replace `<Link />` for the current language with `<span />`.

Reply:
fixing the link is easy, we could just take the pathname from router. So we don't close the drop down menu?

Reply:
makes sense that the current language is  not clickable 👍

Reply:
So we just want to use a span which has no href. And when you click it it closes the popup.

Reply:
maybe we shouldn't even close the popup

Reply:
But then the user wouldn't get any feedback

Reply:
right, done

Reply:
@Schedulaar we don't need to check if closeDropDownCallback is defined, because if it's undefined nothing happens which is what we want